google/cloud/spanner/samples/client_samples.cc

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/spanner/admin/database_admin_client.h"
+#include "google/cloud/spanner/admin/instance_admin_client.h"
+#include "google/cloud/spanner/client.h"
+#include "google/cloud/spanner/testing/random_database_name.h"
+#include "google/cloud/spanner/testing/random_instance_name.h"
+#include "google/cloud/credentials.h"
+#include "google/cloud/internal/getenv.h"
+#include "google/cloud/internal/random.h"
+#include "google/cloud/testing_util/example_driver.h"
+#include <fstream>
+#include <string>
+#include <vector>
+
+namespace {
+
+void ClientSetEndpoint(std::vector<std::string> const& argv) {
+  if (argv.size() != 3) {
+    throw google::cloud::testing_util::Usage{
+        "client-set-endpoint <project-id> <instance-id> <database-id>"};
+  }
+  //! [client-set-endpoint]
+  namespace spanner = ::google::cloud::spanner;
+  [](std::string const& project_id, std::string const& instance_id,
+     std::string const& database_id) {
+    auto options = google::cloud::Options{}.set<google::cloud::EndpointOption>(
+        "private.googleapis.com");
+    return spanner::Client(spanner::MakeConnection(
+        spanner::Database(project_id, instance_id, database_id), options));
+  }
+  //! [client-set-endpoint]
+  (argv.at(0), argv.at(1), argv.at(2));
+}
+
+void ClientWithServiceAccount(std::vector<std::string> const& argv) {
```

Comment:
and this to `WithServiceAccount / [with-service-account]`

Reply:
Fixed